### PR TITLE
Expand ca bundle injection enhancement to cover all supported types

### PR DESCRIPTION
**Pull request by marun**
Also addresses comments from https://github.com/openshift/enhancements/pull/184

/cc @deads2k @sttts

### CODE REVIEW

File: enhancements/authentication/service-ca-bundle-injection.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    annotation.
+
+- Supported resource types:
+  - `APIService` (`apiregistration.k8s.io`)
+    - Injection field: `spec.caBundle`
+    - Supported annotations: alpha,beta
+  - `ConfigMap`
+    - Injection field: `data["service-ca.crt"]`
+    - Supported annotations: alpha,beta
+  - `CustomResourceDefinition` (`apiextensions.k8s.io`)
```
Comment (deads2k) [inline enhancements/authentication/service-ca-bundle-injection.md]: does this represent current state or something you're trying to add?

File: enhancements/authentication/service-ca-bundle-injection.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  - Older types may support both alpha and beta annotations.
+  - Newer types added in 4.4 and above should only respond to the beta
+    annotation.
+
+- Supported resource types:
+  - `APIService` (`apiregistration.k8s.io`)
+    - Injection field: `spec.caBundle`
+    - Supported annotations: alpha,beta
+  - `ConfigMap`
+    - Injection field: `data["service-ca.crt"]`
```
Comment (deads2k) [inline enhancements/authentication/service-ca-bundle-injection.md]: huh, really?  not ca-bundle.crt?

File: enhancements/authentication/service-ca-bundle-injection.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  - `MutatingWebhookConfiguration` (`admissionregistration.k8s.io`)
+    - Injection field: `webhooks[].clientConfig.caBundle`
+    - Supported annotations: beta
+    - Notes: See below
+  - `ValidatingWebhookConfiguration` (`admissionregistration.k8s.io`)
+    - Injection field: `webhooks[].clientConfig.caBundle`
+    - Supported annotations: beta
+    - Notes: See below
+
+- Injection for `{Mutating|Validating}WebhookConfiguration` resources
```
Comment (deads2k) [inline enhancements/authentication/service-ca-bundle-injection.md]: nit: sub-section on Mutating and Validating WebhookConfiguration

File: enhancements/authentication/service-ca-bundle-injection.md (hunk 2 above)
Comment (marun) [reply enhancements/authentication/service-ca-bundle-injection.md]: Yup.

File: enhancements/authentication/service-ca-bundle-injection.md (hunk 1 above)
Comment (marun) [reply enhancements/authentication/service-ca-bundle-injection.md]: This is the current state.

File: enhancements/authentication/service-ca-bundle-injection.md (hunk 3 above)
Comment (marun) [reply enhancements/authentication/service-ca-bundle-injection.md]: I'm not sure what you're asking for, can you elaborate?

**Pull request merged**